apps/web/src/hooks/useBestAMMTrade.ts

```diff
@@ -67,33 +75,117 @@ export function useBestAMMTrade({ type = 'quoter', ...params }: useBestAMMTradeO
 
   const isQuoterAPIEnabled = useMemo(() => Boolean(!isWrapping && type === 'api'), [isWrapping, type])
   const isIceQuoterAPIEnabled = useMemo(() => Boolean(!isWrapping && type === 'auto'), [isWrapping, type])
+  const isAkkaQuoterAPIEnabled = useMemo(() => Boolean(!isWrapping && type === 'auto'), [isWrapping, type])
 
-  const apiAutoRevalidate = typeof autoRevalidate === 'boolean' ? autoRevalidate : isQuoterAPIEnabled || isIceQuoterAPIEnabled
+  const apiAutoRevalidate =
+    typeof autoRevalidate === 'boolean' ? autoRevalidate : isQuoterAPIEnabled || isIceQuoterAPIEnabled
 
   // switch to api when it's stable
-  let bestTradeFromIceQuoterApi = useBestAMMTradeFromQuoterApi({
+  const bestTradeFromIceQuoterApi = useBestAMMTradeFromQuoterApi({
     ...params,
     enabled: Boolean(enabled && isIceQuoterAPIEnabled),
     autoRevalidate: apiAutoRevalidate,
   })
+
+  const bestTradeFromAkkaQuoterApi = useBestAMMTradeFromAkkaQuoterApi({
+    ...params,
+    enabled: Boolean(enabled && isAkkaQuoterAPIEnabled),
+    autoRevalidate: apiAutoRevalidate,
+  })
+
   const bestTradeFromQuoterApi = useBestAMMTradeFromQuoterWorker2({
     ...params,
     enabled: Boolean(enabled && (isQuoterAPIEnabled || isIceQuoterAPIEnabled)),
     autoRevalidate: apiAutoRevalidate,
   })
-  let quoterBetterThanAPI = false
-  if (enabled && isIceQuoterAPIEnabled) {
-    if (!bestTradeFromQuoterApi.isLoading && bestTradeFromQuoterApi.trade?.routes) {
-      if (!bestTradeFromIceQuoterApi || !bestTradeFromIceQuoterApi.trade?.routes) {
-        quoterBetterThanAPI = true
-      } else if(bestTradeFromQuoterApi.trade.outputAmount.greaterThan(bestTradeFromIceQuoterApi.trade.outputAmount)) {
-        quoterBetterThanAPI = true
+
+  type Result = 'AKKA' | 'ICE' | 'Quoter'
+  let betterResult: Result
+
+  if (enabled && (isIceQuoterAPIEnabled || isAkkaQuoterAPIEnabled)) {
+    if (query.referrer === 'AKKA') {
+      betterResult = 'AKKA'
+    } else {
+      if (
+        !bestTradeFromQuoterApi.isLoading &&
+        !bestTradeFromAkkaQuoterApi.isLoading &&
+        !bestTradeFromIceQuoterApi.isLoading
+      ) {
+        if (!bestTradeFromQuoterApi.trade && !bestTradeFromAkkaQuoterApi.trade && !bestTradeFromIceQuoterApi.trade) {
```

Comment:
Please do this huge if else block in a different way. Best would be to make it a list of quoters and get the best one. You can get the output amount of each quoter and if it is not available, simply set it to none. Then you filter all output amounts whether they are none. If the filtered list is empty, use UI quoter, else use the one with the biggest number of the list.

Reply:
Can you please explane more?